docs/src/pages/getting-started/astro.mdx

```diff
@@ -1 +1,229 @@
-# Astro (Coming Soon!)
+import { Callout, Steps, Tab, Tabs } from "nextra-theme-docs";
+
+import EnvSetup from "./_shared/env-setup.mdx";
+import TailwindSetup from "./_shared/tailwind-setup.mdx";
+
+# Astro Setup
+
+You can use Astro with any of the frontend frameworks we support. This guide
+will walk you through setting it up using React, but the steps will be near
+identical for other frameworks too.
+
+<Callout type="info">
+  Check out a full example
+  [here](https://github.com/pingdotgg/uploadthing/tree/main/examples/minimal-astro-react)
+</Callout>
+
+<EnvSetup />
+
+## Set Up A FileRouter
+
+<Steps>
+### Creating your first FileRoute
+
+All files uploaded to uploadthing are associated with a FileRoute. Think of a
+FileRoute similar to an endpoint, it has:
+
+- Permitted types ["image", "video", etc]
+- Max file size
+- (Optional) `middleware` to authenticate and tag requests
+- `onUploadComplete` callback for when uploads are completed
+
+```ts copy filename="src/server/uploadthing.ts"
+import { createUploadthing, type FileRouter } from "uploadthing/server";
+
+const f = createUploadthing();
+
+const auth = (req: Request) => ({ id: "fakeId" }); // Fake auth function
+
+// FileRouter for your app, can contain multiple FileRoutes
+export const ourFileRouter = {
+  // Define as many FileRoutes as you like, each with a unique routeSlug
+  imageUploader: f({ image: { maxFileSize: "4MB" } })
+    // Set permissions and file types for this FileRoute
+    .middleware(async ({ req }) => {
+      // This code runs on your server before upload
+      const user = await auth(req);
+
+      // If you throw, the user will not be able to upload
+      if (!user) throw new Error("Unauthorized");
+
+      // Whatever is returned here is accessible in onUploadComplete as `metadata`
+      return { userId: user.id };
+    })
+    .onUploadComplete(async ({ metadata, file }) => {
+      // This code RUNS ON YOUR SERVER after upload
+      console.log("Upload complete for userId:", metadata.userId);
+
+      console.log("file url", file.url);
+
+      // !!! Whatever is returned here is sent to the clientside `onClientUploadComplete` callback
+      return { uploadedBy: metadata.userId };
+    }),
+} satisfies FileRouter;
+
+export type OurFileRouter = typeof ourFileRouter;
+```
+
+### Create a Astro API route using the FileRouter
+
+<Callout type="info">
+  File path here doesn't matter, you can serve this from any route. We recommend
```

Comment:
I would flip this to "We recommend serving the UploadThing endpoint out of /api/uploadthing, but you [can change the route if you like]" with a link to the custom endpoint doc

Reply:
just copied from next docs

Reply:
i wont change it in this PR, do a follow-up changing all of them instead